### PR TITLE
Refactored styling of category page (#617)

### LINKED ISSUE #617: Simplify styling of category page
## What is the motivation for adding/enhancing this feature?
Refactor styling of category page

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [X] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by azkyakhan**
### Related Issues
#617 

Closes #617 

### Short Description and Why It's Useful
Refactored styling of category page

### Screenshots of Visual Changes before/after (If There Are Any)
category-page-desktop(1)-before
category-page-desktop(2)-before
category-page-desktop(3)-before
category-page-desktop(1)-after
category-page-desktop(2)-after
category-page-desktop(3)-after
category-page-mobile(1)-before
category-page-mobile(2)-before
category-page-mobile(1)-after
category-page-mobile(2)-after

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [X] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: pages/Category.vue [hunk 1]
```diff
@@ -46,26 +51,26 @@
...
           <SfButton
-            class="sf-button--text navbar__filters-button sort-by__button mobile-only"
+            class="sf-button--text navbar__button mobile-only"
             @click="$refs.SortBy.toggle()"
           >
             {{ $t('Sort By') }}
             <ASortIcon />
           </SfButton>
         </div>
+
```
Comment (dt2patel) [inline pages/Category.vue:61]: I think we can add a comment here that if a user wants to add additional items to the nabber they should insert them right here, before the counter.

File: pages/Category.vue [hunk 2]
```diff
@@ -46,26 +51,26 @@
...
         <div class="navbar__counter">
-          <span class="navbar__label desktop-only">
+          <span class="label desktop-only">
             {{ $t("Products found") }}:
           </span>
           <strong class="desktop-only">{{ getCategoryProductsTotal }}</strong>
-          <span class="navbar__label mobile-only">
+          <span class="count mobile-only">
```
Comment (dt2patel) [inline pages/Category.vue:67]: maybe we could add a space here to make it easy to differentiate the mobile and desktop UI. maybe even annotate them with comments?

File: pages/Category.vue [hunk 3]
```diff
@@ -604,127 +613,123 @@ export default {
...
+
   &__counter {
-    font-family: var(--font-family-secondary);
-    grid-column: 3;
-    justify-self: end;
     grid-area: counter;
+    justify-self: end;
   }
+
   &__view {
```
Comment (dt2patel) [inline pages/Category.vue:707]: Since this class isn't used rn but will be useful if we implement a grid/list view switcher, it'll make it easier for new people if this is annotated and commented out?

File: pages/Category.vue (hunk 1 above)
Comment (azkyakhan) [reply pages/Category.vue:61]: Ok I will do that

File: pages/Category.vue (hunk 2 above)
Comment (azkyakhan) [reply pages/Category.vue:67]: Ok

File: pages/Category.vue (hunk 3 above)
Comment (azkyakhan) [reply pages/Category.vue:707]: Ok

File: pages/Category.vue [hunk 4]
```diff
@@ -604,127 +613,123 @@ export default {
...
+    border: 1px solid var(--c-light);
+    border-width: 1px 1px 1px 0;
+  }
+
 .navbar {
-  position: relative;
-  display: flex;
+  grid-area: navbar;
+  display: grid;
+  grid:"filter counter sort" / 1fr minmax( auto, max-content) 1fr;
```
Comment (dt2patel) [inline pages/Category.vue:642]: I think a comment here that describes how to expand this grid during customization in plain english would be helpful. Something like "To add more items to the navbar, add their grid-area to the template. The mobile navbar shouldn't have more than 3 items at a time, so replace rather than add."
